### PR TITLE
[WIP] More tests, various fixes

### PR DESCRIPTION
**Pull request by frdwrd**
Do not merge. Still a work in progress.

I'm a little worried that I messed up merging in master. Can I get a careful code review?

### CODE REVIEW

**Comment by frdwrd** [comment]
Still don't merge. :)

Feedback needed on test_file_download. I have a bunch of stuff commented out that needs work.

File: storj/http.py [hunk 1]
```diff
@@ -41,7 +40,7 @@ class Client(object):
         public_key_hex ():
     """
 
-    logger = logging.getLogger('%s.Client' % __name__)
+    logger = logging.getLogger(web_socket.Client.__name__)
```
Comment (frdwrd) [inline storj/http.py]: I need to learn more about the logger. I changed this line when I changed the import statement from `from .web_socket import Client` to `from storj import web_socket`. I was basically applying blanket changes to everywhere I saw Client.

I'll make the change suggested.

File: tests/unit/http_test.py [hunk 2]
```diff
@@ -282,32 +295,78 @@ def test_key_dump(self, mock_stdout):
...
+        self.client.key_export()
+
+        file_write_calls = [
+            mock.call('7'),
+            mock.call('8')]
+        print_calls = [
+            mock.call('Writing your public key to file...'),
+            mock.call('\n'),
+            mock.call('Writing private key to file... Keep this secret!'),
+            mock.call('\n'),
```
Comment (steenzout) [inline tests/unit/http_test.py]: since you are using `any_calls` you don't need to repeat this.

**Comment by frdwrd** [comment]
Dang. Forgot to check py35 locally before pushing the commit.

File: storj/http.py [hunk 3]
```diff
@@ -451,7 +451,7 @@ def key_export(self):
         with open('private.pem', 'wb') as keyfile:
             keyfile.write(self.private_key.to_pem())
 
-        print("Wrote keyfiles to dir: " + os.getcwd())
+        print('Wrote keyfiles to dir: ' + os.getcwd())
```
Comment (steenzout) [inline storj/http.py]: Better and faster
```
print('Wrote keyfiles to dir %s: ' % os.getcwd())
```

**Comment by frdwrd** [comment]
Made both recommended changes. Is string formatting always faster than concatenation?

**Comment by steenzout** [comment]
linking #14 to this issue.

**Comment by steenzout** [comment]
@frdwrd yes I think so.

note when writing log messages you should build the messages the following way:
```
self.logger.info('parameter x=%s y=%s', x, y)
# not 'parameter x=%s y=%s' % (x, y)
```

File: tox.ini [hunk 4]
```diff
@@ -34,9 +34,11 @@ commands =
         --aggressive \
         --recursive \
         --max-line-length 119 \
+        --ignore=W503 \
         storj tests
-    pep8 tests
-
+    pep8 \
+        --ignore=W503 \
```
Comment (steenzout) [inline tox.ini]: oversight from my part.

**Comment by frdwrd** [comment]
This is ready to merge afaik.

**Pull request merged**